sympy/vector/coordsysrect.py

```diff
@@ -1,19 +1,45 @@
+import types
+from sympy.simplify import simplify
 from sympy.core.basic import Basic
 from sympy.vector.scalar import BaseScalar
-from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol
+from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol, symbols, sqrt, Lambda, Dummy
 from sympy.core.compatibility import string_types, range
 from sympy.core.cache import cacheit
 from sympy.vector.orienters import (Orienter, AxisOrienter, BodyOrienter,
                                     SpaceOrienter, QuaternionOrienter)
-import sympy.vector
 
 
-class CoordSysCartesian(Basic):
+def build_instance_from_lambda(fun=lambda x, y, z: (x, y, z)):
```

Comment:
`CoordSystem` is already taken in `sympy.diffgeom`. Anyway, it's better to jot down some API proposal and ask @moorepants what he thinks. He's the leading developer of the vector module.